### ISSUE TITLE
machine unregistered

### ISSUE THREAD

**Issue by lucasart** [AndyGrant/OpenBench#38]
Tried to run the client, but it crashed:
```
[lucas@megatron Client]$ python3 ./OpenBench.py -U lucasart -P ********* -S http://chess.grantnet.us/ -T 7
<Warning> Machine unregistered, will register with Server
Downloading : <!DOCTYPE html><html lang="en"><head><meta content="text/html; charset=utf-8" http-equiv="content-type"/><title>Page not found at //getFiles/</title><meta content="NONE,NOARCHIVE" name="robots"/><style type="text/css">html * { padding:0; margin:0; } body * { padding:10px 20px; } body * * { padding:0; } body { font:small sans-serif; background:#eee; color:#000; } body>div { border-bottom:1px solid #ddd; } h1 { font-weight:normal; margin-bottom:.4em; } h1 span { font-size:60%; color:#666; font-weight:normal; } table { border:none; border-collapse: collapse; width:100%; } td, th { vertical-align:top; padding:2px 3px; } th { width:12em; text-align:right; color:#666; padding-right:.5em; } #info { background:#f6f6f6; } #info ol { margin: 0.5em 4em; } #info ol li { font-family: monospace; } #summary { background: #ffc; } #explanation { background:#eee; border-bottom: 0px none; }</style></head><body><div id="summary"><h1>Page not found <span>(404)</span></h1><table class="meta"><tbody><tr><th>Request Method:</th><td>GET</td></tr><tr><th>Request URL:</th><td>http://chess.grantnet.us//getFiles/</td></tr></tbody></table></div><div id="info"><p>Using the URLconf defined in <code>OpenSite.urls</code>, Django tried these URL patterns, in this order:</p><ol><li>admin/</li><li>register/</li><li>login/</li><li>logout/</li><li>viewProfile/</li><li>editProfile/</li><li>index/</li><li>index/&lt;int:page&gt;/</li><li>greens/</li><li>greens/&lt;int:page&gt;/</li><li>search/</li><li>viewUser/&lt;str:username&gt;/</li><li>viewUser/&lt;str:username&gt;/&lt;int:page&gt;/</li><li>users/</li><li>machines/</li><li>eventLog/</li><li>eventLog/&lt;int:page&gt;/</li><li>newTest/</li><li>viewTest/&lt;int:id&gt;/</li><li>editTest/&lt;int:id&gt;/</li><li>approveTest/&lt;int:id&gt;/</li><li>restartTest/&lt;int:id&gt;/</li><li>stopTest/&lt;int:id&gt;/</li><li>deleteTest/&lt;int:id&gt;/</li><li>getFiles/</li><li>getWorkload/</li><li>wrongBench/</li><li>submitNPS/</li><li>submitResults/</li><li></li></ol><p>The current path, <code>/getFiles/</code>, didn't match any of these.</p></div><div id="explanation"><p>You're seeing this error because you have <code>DEBUG = True</code> in your Django settings file. Change that to <code>False</code>, and Django will display a standard 404 page.</p></div></body></html>cutechess-linux
Traceback (most recent call last):
  File "./OpenBench.py", line 418, in <module>
    getCoreFiles()
  File "./OpenBench.py", line 74, in getCoreFiles
    getFile(location + 'cutechess-linux', 'cutechess')
  File "./OpenBench.py", line 54, in getFile
    request = requests.get(url=source, stream=True, timeout=HTTP_TIMEOUT)
  File "/usr/lib/python3.8/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 640, in send
    adapter = self.get_adapter(url=request.url)
  File "/usr/lib/python3.8/site-packages/requests/sessions.py", line 731, in get_adapter
    raise InvalidSchema("No connection adapters were found for '%s'" % url)
requests.exceptions.InvalidSchema: No connection adapters were found for '<!DOCTYPE html><html lang="en"><head><meta content="text/html; charset=utf-8" http-equiv="content-type"/><title>Page not found at //getFiles/</title><meta content="NONE,NOARCHIVE" name="robots"/><style type="text/css">html * { padding:0; margin:0; } body * { padding:10px 20px; } body * * { padding:0; } body { font:small sans-serif; background:#eee; color:#000; } body>div { border-bottom:1px solid #ddd; } h1 { font-weight:normal; margin-bottom:.4em; } h1 span { font-size:60%; color:#666; font-weight:normal; } table { border:none; border-collapse: collapse; width:100%; } td, th { vertical-align:top; padding:2px 3px; } th { width:12em; text-align:right; color:#666; padding-right:.5em; } #info { background:#f6f6f6; } #info ol { margin: 0.5em 4em; } #info ol li { font-family: monospace; } #summary { background: #ffc; } #explanation { background:#eee; border-bottom: 0px none; }</style></head><body><div id="summary"><h1>Page not found <span>(404)</span></h1><table class="meta"><tbody><tr><th>Request Method:</th><td>GET</td></tr><tr><th>Request URL:</th><td>http://chess.grantnet.us//getFiles/</td></tr></tbody></table></div><div id="info"><p>Using the URLconf defined in <code>OpenSite.urls</code>, Django tried these URL patterns, in this order:</p><ol><li>admin/</li><li>register/</li><li>login/</li><li>logout/</li><li>viewProfile/</li><li>editProfile/</li><li>index/</li><li>index/&lt;int:page&gt;/</li><li>greens/</li><li>greens/&lt;int:page&gt;/</li><li>search/</li><li>viewUser/&lt;str:username&gt;/</li><li>viewUser/&lt;str:username&gt;/&lt;int:page&gt;/</li><li>users/</li><li>machines/</li><li>eventLog/</li><li>eventLog/&lt;int:page&gt;/</li><li>newTest/</li><li>viewTest/&lt;int:id&gt;/</li><li>editTest/&lt;int:id&gt;/</li><li>approveTest/&lt;int:id&gt;/</li><li>restartTest/&lt;int:id&gt;/</li><li>stopTest/&lt;int:id&gt;/</li><li>deleteTest/&lt;int:id&gt;/</li><li>getFiles/</li><li>getWorkload/</li><li>wrongBench/</li><li>submitNPS/</li><li>submitResults/</li><li></li></ol><p>The current path, <code>/getFiles/</code>, didn't match any of these.</p></div><div id="explanation"><p>You're seeing this error because you have <code>DEBUG = True</code> in your Django settings file. Change that to <code>False</code>, and Django will display a standard 404 page.</p></div></body></html>cutechess-linux'
```

**Comment by AndyGrant** [comment]
Duplicate of https://github.com/AndyGrant/OpenBench/issues/35

You need to delete the trailing``/`` in the server name. Then you need to call me a fool for doing URL manipulations as raw text.

Would you like me to add Demolito to OpenBench? I will need to get Noobpwnftw to update his client file. You don't have quite the setup OpenBench expects, but thats okay because the new client is about generalizing all of it.

**Comment by lucasart** [comment]
I think it works now, but hard to say for sure, as there's no work to do:
```
[lucas@megatron Client]$ python3 ./Client.py -U lucasart -P utvmq2tla -S http://chess.grantnet.us -T 7
[NOTE] Machine Is Unregistered
Downloading http://github.com/AndyGrant/OpenBench/raw/master/CoreFiles/cutechess-linux
Downloading http://github.com/AndyGrant/OpenBench/raw/master/CoreFiles/libcutechess.so.1
[NOTE] Server Has No Work
```
Can you validate the test I created ? It should work with the current client (I think).

PS: New client is Client.py and old one is OpenBench.py, right ?

**Comment by AndyGrant** [comment]
I was able to run 5 games on my client with success, but Noobpwn's client threw invalid bench (which is more likely could not compile, or could not fetch the repo). Ill reach out to him, since I need to finalize Client.py

**Comment by lucasart** [comment]
ok. thanks.

PS: The trailing `/` was copy pasted from your README file. you may want to correct that.

**Comment by lucasart** [comment]
Here's my proposal to improve the README file on the client usage part. I think this is the most important part, where the barrier of entry should be the lowest, because you want to have more people donate CPU resources.

# Setting Up The Client For Linux

## Install Python3 and GNU C/C++ compiler

- Arch based: `pacman -S python gcc`
- Debian based: `apt-get install python3 gcc g++`
- RedHat based: `yum install python3 gcc g++`

## Download the client

`curl https://raw.githubusercontent.com/AndyGrant/OpenBench/master/Client/OpenBench.py -o ./OpenBench.py`

# Running The Client

The client takes four arguments: Username, Password, Server, Threads. First create an account via the instance's webpage. The instance will also be the Server provided to the client. 
Threads tells the Client how many games can be run in parallel. This should be no more than your CPU count minus one. For hyperthreaded machines, where hyperthreading support is strong, 
you may use all threads provided by CPU minus one cores. The following would connect to the main instance using 4 threads or CPUs.

``python3 OpenBench.py -U username -P password -S http://chess.grantnet.us -T 4``

**Comment by lucasart** [comment]
Note that Demolito's makefile requires git. I am not keen to remove this dependency, because my automatic versioning and appveyor setup depend on it. Once you got used to automatic versioning, you really don't want to go back to updating version numbers manually anymore…

**Comment by lucasart** [comment]
closing. issue solved (new client).
